CONTRIBUTING.md

```diff
@@ -23,34 +23,62 @@ If you find a bug anywhere, you can fix it.
 
 # Setup
 
-1. Fork this repository.
+1. Fork this repository using the **Fork** option at the top-right corner of this page. This will create your own copy of this repository. You'll be redirected to your forked repository. Copy the link of this repository (which will look like `https://github.com/<your-username>/website-welcome/`) as you'll need it in the step 2.
 
-2. Clone your forked repository.
+![how-to-fork](https://i.imgur.com/VfoTxmy.png)
+
+2. Clone your forked repository, this will download your copy of repository in your computer. To do this, open your terminal (command prompt/bash/git bash) and enter the following command, paste your link after the word **clone** without the **<>**.
 ``` 
-git clone https://github.com/<your-username>/website-welcome/ 
+git clone <link which you copied in the step 1>
 ```
 
-3. Add the original repository as a remote repository.
+3. Add the original repository as a remote repository, so that you can anytime pull the latest changes from the main repository which is being deployed, this needs to be done only for the first time.
 ``` 
-git remote add https://github.com/Real-Dev-Squad/website-welcome/ 
+git remote add upstream https://github.com/Real-Dev-Squad/website-welcome/ 
+```
+To make sure you always have the latest copy of the main repository before starting with your changes, execute the following command:
+```
+git pull upstream main
 ```
 
-4. Create a new branch to work on.
+4. Create a new branch to work on. We require a different branch so that we always have a stable, working version in the default (main) branch.
 ``` 
 git checkout -b <branch-name> 
 ```
 
-5. Do your changes.
+For example, if I want to name my branch as **develop**, I'll enter the following command:
+``` 
+git checkout -b develop
+```
+
+5. Perform the tasks you wanted to, can be anything, ranging from fixing simple typo to re-designing the whole page!
 
-6. Commit them.
+6. Now you have made the changes, though they are saved in your system, Git doesn't know what changes you've done. So you have **commit** your changes. First step is to add the files which you want to add to the staging area, the dot after **add** in the first command tells Git to check for changes in all the files. The second step is about committing your changes. The message part is short description of your commit, like "fixed the css for homepage".
```

Comment:
you have _to_ **commit**

Reply:
Fix css for homepage.
The convention for git commits is to imperative mood in the subject line.
Ex - Merge branch.
Revert commit.